apps/client/src/context/AuthContext.jsx

```diff
@@ -0,0 +1,73 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import { apiFetch } from "../utils/api";
+
+const AuthCtx = createContext(null);
+export const useAuth = () => useContext(AuthCtx);
+
+export default function AuthProvider({ children }) {
+  const [user, setUser] = useState(null);
+  const [token, setToken] = useState(() => localStorage.getItem("token"));
+
+  // restore user on page load
+  useEffect(() => {
+    const userStr = localStorage.getItem("user");
+    if (token && userStr) {
+      try {
+        setUser(JSON.parse(userStr));
+      } catch {}
+    }
+  }, [token]);
+
+  async function register({
+    email,
+    password,
+    name,
+    employeeId,
+    sterilizationNumber,
+  }) {
+    const r = await apiFetch("/api/auth/register", {
+      method: "POST",
+      body: JSON.stringify({
+        email,
+        password,
+        name,
+        employeeId,
+        sterilizationNumber,
```

Comment:
All api functions should be stored in `api.js` and (or) `auth.js`, to separate the api logic from the components

Reply:
Please, move all api functions into `api.js` or `auth.js`, to separate the logic